tests/cloner_test.go

```diff
@@ -84,6 +84,88 @@ var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:compo
 		sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile)
 		cloneOfAnnoExistenceTest(f, f.Namespace.Name)
 	})
+
+	It("Should clone across nodes when multiple local volumes exist,", func() {
+		// Get nodes, need at least 2
+		nodeList, err := f.K8sClient.CoreV1().Nodes().List(metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		if len(nodeList.Items) < 2 {
+			Skip("Need at least 2 nodes to copy accross nodes")
+		}
+		nodeMap := make(map[string]bool)
+		for _, node := range nodeList.Items {
+			if ok, _ := nodeMap[node.Name]; !ok {
+				nodeMap[node.Name] = true
+			}
+		}
+		// Find PVs and identify local storage, the PVs should already exist.
+		pvList, err := f.K8sClient.CoreV1().PersistentVolumes().List(metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		var sourcePV, targetPV *v1.PersistentVolume
+		var storageClassName string
+		// Find the source and target PVs so we can label them.
+		for _, pv := range pvList.Items {
+			if sourcePV == nil {
+				if pv.Spec.StorageClassName != "" {
+					storageClassName = pv.Spec.StorageClassName
+				}
+				pvNode := pv.Spec.NodeAffinity.Required.NodeSelectorTerms[0].MatchExpressions[0].Values[0]
+				if ok, val := nodeMap[pvNode]; ok && val {
+					nodeMap[pvNode] = false
+					By("Labeling PV " + pv.Name + " as source")
+					sourcePV = &pv
+					if sourcePV.GetLabels() == nil {
+						sourcePV.SetLabels(make(map[string]string))
+					}
+					sourcePV.GetLabels()["source-pv"] = "yes"
```

Comment:
are labels guaranteed to be non nil?

Reply:
I made sure right above that they are not nil
```
if sourcePV.GetLabels() == nil {
	sourcePV.SetLabels(make(map[string]string))
}
```

Reply:
LOL

Reply:
what's up with the crio failure?

Reply:
No idea, haven't had a chance to look at what is failing yet.

Reply:
The crio lane has a non local volume PV without a NodeAffinity which is causing the npe. Fixed.